components/Icon/index.test.jsx

```diff
@@ -0,0 +1,25 @@
+import { render, screen } from '@testing-library/react';
+
+import Icon from '.';
+
+describe('Icon', () => {
+  it('renders the provided icon', () => {
+    const props = {
+      src: '/icon.svg',
+      alt: 'icon',
+    };
+
+    const dataTestId = 'icon';
+
+    render(
+      <div data-test-id={dataTestId}>
+        <Icon {...props} />
+      </div>,
+    );
+
+    const icon = screen.getByTestId(dataTestId);
+
+    expect(icon).toBeVisible();
+    expect(icon).toHaveTextContent(props.src);
```

Comment:
```suggestion
    const image = icon.getByRole('img');

    expect(image).toHaveAttribute('src', props.src);
```
Should we be more specific to ensure the src is not contained in an error message? (e.g. if the image component would raise an error "the image path ${props.src} is invalid", this test would still pass.

Though my suggestion has a limitation: we assume things about the image component while we do not have direct control over it (as it comes from `next/image`)

Reply:
Re-wrote `next/image` mock to test image by `src` attribute in 30b5d8340cd5b12952568fa9b4f326451f3c58ed